core/store/src/trie/shard_tries.rs

```diff
@@ -483,6 +483,36 @@ impl ShardTries {
             memtries.write().unwrap().delete_until_height(height);
         }
     }
+
+    /// Freezes in-memory trie for source shard and copies reference to it to
+    /// target shards.
+    /// Needed to serve queries for these shards just after resharding, before
+    /// proper memtries are loaded.
+    pub fn freeze_mem_tries(
+        &self,
+        source_shard_uid: ShardUId,
+        target_shard_uids: Vec<ShardUId>,
```

Comment:
nit: Are those the parent and children? If so would it make sense to name those that?